src/com/carolinarollergirls/scoreboard/xml/ScoreBoardXmlListener.java

```diff
@@ -175,7 +176,62 @@ public void scoreBoardChange(ScoreBoardEvent event) {
+			Element e = getStatsElement();
+			if (prop.equals(Stats.EVENT_REMOVE_PERIOD)) {
+				Stats.PeriodStats ps = (Stats.PeriodStats)(event.getValue());
+				if (isPersistent()) {
+					editor.removeElement(e, "PeriodStats", String.valueOf(ps.getPeriodNumber()));
+				} else {
+					editor.setRemovePI(converter.toElement(e, ps));
+				}
+			}
+		} else if (p.getProviderName().equals("PeriodStats")) {
+			Element e = getPeriodStatsElement((Stats.PeriodStats)p);
+			if (prop.equals(Stats.PeriodStats.EVENT_REMOVE_JAM)) {
+				Stats.JamStats js = (Stats.JamStats)(event.getValue());
+				if (isPersistent()) {
+					editor.removeElement(e, "JamStats", String.valueOf(js.getJamNumber()));
+				} else {
+					editor.setRemovePI(converter.toElement(e, js));
+				}
+			}
+		} else if (p.getProviderName().equals("JamStats")) {
+			Element e = getJamStatsElement((Stats.JamStats)p);
+			if (prop.equals(Stats.JamStats.EVENT_STATS)) {
+				Stats.JamStats js = (Stats.JamStats)event.getValue();
+				editor.setElement(e, "JamClockElapsedEnd", null, String.valueOf(js.getJamClockElapsedEnd()));
+				editor.setElement(e, "PeriodClockElapsedStart", null, String.valueOf(js.getPeriodClockElapsedStart()));
+				editor.setElement(e, "PeriodClockElapsedEnd", null, String.valueOf(js.getPeriodClockElapsedEnd()));
+				editor.setElement(e, "PeriodClockWalltimeStart", null, String.valueOf(js.getPeriodClockWalltimeStart()));
+				editor.setElement(e, "PeriodClockWalltimeEnd", null, String.valueOf(js.getPeriodClockWalltimeEnd()));
+			}
+		} else if (p.getProviderName().equals("TeamStats")) {
+			Element e = getTeamStatsElement((Stats.TeamStats)p);
+			if (prop.equals(Stats.TeamStats.EVENT_STATS)) {
+				Stats.TeamStats ts = (Stats.TeamStats)event.getValue();
+				editor.setElement(e, "JamScore", null, String.valueOf(ts.getJamScore()));
+				editor.setElement(e, "TotalScore", null, String.valueOf(ts.getTotalScore()));
+				editor.setElement(e, "LeadJammer", null, ts.getLeadJammer());
+				editor.setElement(e, "StarPass", null, String.valueOf(ts.getStarPass()));
+				editor.setElement(e, "Timeouts", null, String.valueOf(ts.getTimeouts()));
+				editor.setElement(e, "OfficialReviews", null, String.valueOf(ts.getOfficialReviews()));
+			} else if (prop.equals(Stats.TeamStats.EVENT_REMOVE_SKATER)) {
+				Stats.SkaterStats ss = (Stats.SkaterStats)(event.getValue());
+				if (isPersistent()) {
+					editor.removeElement(e, "SkaterStats", ss.getSkaterId());
+				} else {
+					editor.setRemovePI(converter.toElement(e, ss));
+				}
+			}
+		} else if (p.getProviderName().equals("SkaterStats")) {
+			Element e = getSkaterStatsElement((Stats.SkaterStats)p);
+			if (prop.equals(Stats.SkaterStats.EVENT_STATS)) {
+				Stats.SkaterStats ss = (Stats.SkaterStats)event.getValue();
+				editor.setElement(e, "Position", null, ss.getPosition());
+				editor.setElement(e, "PenaltyBox", null, String.valueOf(ss.getPenaltyBox()));
+			}
+    } else if (p.getProviderName().equals("Clock")) {
```

Comment:
indentation of this line is wrong. As is the block above the new code ("Skater"), presumably from a previous commit by you (it's related to penalties).

Reply:
As I've previously indicated, I plan on running a pretty printer over the whole codebase once noone has changes pending. The current style is very difficult to work for me with due to how non-standard it is.